source/renderer/app/containers/LoadingPage.js

```diff
@@ -115,34 +127,42 @@ export default class LoadingPage extends Component<InjectedProps> {
           onCheckTheTimeAgain={forceCheckLocalTimeDifference}
           onContinueWithoutClockSyncCheck={ignoreSystemTimeChecks}
           onDownloadLogs={this.handleDownloadLogs}
+          disableDownloadLogs={this.state.disableDownloadLogs}
         />
-        <NotificationMessage
-          icon={successIcon}
-          show={stores.uiNotifications.isOpen(id)}
-          onClose={this.closeNotification}
-          clickToClose
+        <GenericNotification
+          id={DOWNLOAD_LOGS_PROGRESS_NOTIFICATION_ID}
+          show={stores.uiNotifications.isOpen(
+            DOWNLOAD_LOGS_PROGRESS_NOTIFICATION_ID
+          )}
+          actionToListenAndOpen={actions.profile.downloadLogs}
+          actionToListenAndClose={actions.profile.downloadLogsSuccess}
+          openNotification={actions.notifications.open}
+          closeNotification={actions.notifications.closeActiveNotification}
+          icon="spinner"
+          hasEllipsis
+          hasCloseButton
+        >
+          {intl.formatMessage(messages.downloadLogsProgress)}
+        </GenericNotification>
+        <GenericNotification
+          id={DOWNLOAD_LOGS_SUCCESS_NOTIFICATION_ID}
+          duration={DOWNLOAD_LOGS_SUCCESS_DURATION}
+          show={stores.uiNotifications.isOpen(
+            DOWNLOAD_LOGS_SUCCESS_NOTIFICATION_ID
+          )}
+          actionToListenAndOpen={actions.profile.downloadLogsSuccess}
+          actionToListenAndClose={actions.profile.downloadLogs}
+          openNotification={actions.notifications.open}
+          closeNotification={actions.notifications.closeActiveNotification}
```

Comment:
will there ever be any other `openNotification` and `closeNotification` prop values? If not we should not repeat the wire-up boilerplate for each instance 😉

Reply:
@DominikGuzei not sure how to make things differently here, as it's a component, not a container. Maybe using context? These props will always be the same, but the component doesn't have access to the actions.

Reply:
@DominikGuzei @nikolaglumac I don't know any other way of doing this. The `<GenericNotification />` component needs to receive the `open` and `closeActiveNotification` notification actions, but as it's a component, it doesn't have access to the `actions`. It works, but's ugly, as every component using it needs to pass them manually. Any suggestion? Or can we leave it like this?

Reply:
Done, created a container that provides these 2 props.